feat(rfs): new resource to manage template (#9326)
huaweicloud/terraform-provider-huaweicloud: docs/resources/rfs_template.md

---
subcategory: "Resource Formation (RFS)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_rfs_template"
description: |- 

>   Manages a template resource within HuaweiCloud.

Reviewer: ```suggestion
  Manages a HCL template resource within HuaweiCloud.
```

---

# huaweicloud_rfs_template

> Manages a template resource within HuaweiCloud.

Reviewer: ```suggestion
Manages a HCL template resource within HuaweiCloud.
```

  pure Terraform file or a ZIP archive:
    + For pure Terraform files, the file name must end with `.tf` or `.tf.json` and comply with HCL syntax.
    + For archives, only ZIP format is supported. The file name must end with `.zip`. After decompression, the archive
      must not contain `.tfvars` files.

* `version_description` - (Optional, String, NonUpdatable) Specifies the description of the initial template version.

* `template_description` - (Optional, String) Specifies the template description.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The resource ID (also `template_name`).

* `template_id` - The unique ID of the template, randomly generated by the template service.

* `create_time` - The time when the template was created, in RFC3339 format (yyyy-mm-ddTHH:MM:SSZ),
  e.g., `1970-01-01T00:00:00Z`.

* `update_time` - The time when the template was last updated, in RFC3339 format (yyyy-mm-ddTHH:MM:SSZ),
  e.g., `1970-01-01T00:00:00Z`.

* `latest_version_id` - The ID of the latest template version.

* `latest_version_description` - The description of the latest template version.

## Import

> Private hooks can be imported using their `template_name`, e.g.

Reviewer 2: ```suggestion
The template can be imported using their `template_name`, e.g.
```

  version_description  = var.version_description 
  template_body        = <<-EOF
variable "vpc_name" {
  type    = string
  default = "my-vpc"
}

resource "huaweicloud_vpc" "vpc" {
  name = var.vpc_name
  cidr = "172.16.0.0/16"
}
EOF
}
```

## Argument Reference

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource. If omitted, the
  provider-level region will be used. Changing this will create a new resource.

* `template_name` - (Required, String, NonUpdatable) Specifies the template name. The name must be unique and can
  contain only letters, digits, hyphens (-), and underscores (_).

* `template_body` - (Optional, String, NonUpdatable) Specifies the Terraform template body. This field
  and `template_uri` are mutually exclusive, one of them must be specified. The template body must be a valid Terraform
  configuration in HCL format.

* `template_uri` - (Optional, String, NonUpdatable) Specifies the OBS URL where the Terraform template is stored. This
  field and `template_body` are mutually exclusive, one of them must be specified. The corresponding file should be a
  pure Terraform file or a ZIP archive:

>     + For pure Terraform files, the file name must end with `.tf` or `.tf.json` and comply with HCL syntax.

Reviewer 3: ```suggestion
  + For pure Terraform files, the file name must end with `.tf` or `.tf.json` and comply with HCL syntax.
```
